core/src/autogluon/core/models/abstract/abstract_model.py

```diff
@@ -1167,6 +1175,141 @@ def load(cls, path: str, reset_paths: bool = True, verbose: bool = True):
                 model.model = model._compiler.load(path=path)
         return model
 
+    def save_learning_curves(self, metrics: str | List[str], curves: dict[dict[str : List[float]]], path: str = None) -> str:
+        """
+        Saves learning curves to disk.
+
+        Outputted Curve Format:
+            out = [
+                metrics,
+                [
+                    [ # log_loss
+                        [0.693147, 0.690162, ...], # train
+                        [0.693147, 0.690162, ...], # val
+                        [0.693147, 0.690162, ...], # test
+                    ],
+                    [ # accuracy
+                        [0.693147, 0.690162, ...], # train
+                        [0.693147, 0.690162, ...], # val
+                        [0.693147, 0.690162, ...], # test
+                    ],
+                    [ # f1
+                        [0.693147, 0.690162, ...], # train
+                        [0.693147, 0.690162, ...], # val
+                        [0.693147, 0.690162, ...], # test
+                    ],
+                ]
+            ]
+
+        Parameters
+        ----------
+        metrics : str or list(str)
+            List of all evaluation metrics computed at each iteration of the curve
+        curves : dict[dict[str : list[float]]]
+            Dictionary of evaluation sets and their learning curve dictionaries.
+            Each learning curve dictionary contains evaluation metrics computed at each iteration.
+            e.g.
+                curves = {
+                        "train": {
+                            'logloss': [0.693147, 0.690162, ...],
+                            'accuracy': [0.500000, 0.400000, ...],
+                            'f1': [0.693147, 0.690162, ...]
+                        },
+                        "val": {...},
+                        "test": {...},
+                    }
+
+        path : str, default None
+            Path where the learning curves are saved, minus the file name.
+            This should generally be a directory path ending with a '/' character (or appropriate path separator value depending on OS).
+            If None, self.path is used.
+            The final curve file is typically saved to os.path.join(path, curves.json).
+
+        Returns
+        -------
+        path : str
+            Path to the saved curves, minus the file name.
+        """
+        if path is None:
```

Comment:
Don't let the save operation occur if learning curves aren't marked as supported.

```python
if not self._get_class_tags().get("supports_learning_curves", False):
    raise AssertionError(...)
```